velox/connectors/hive/storage_adapters/abfs/RegisterAbfsFileSystem.cpp

```diff
@@ -14,16 +14,32 @@
  * limitations under the License.
  */
 
+#include "velox/connectors/hive/storage_adapters/abfs/RegisterAbfsFileSystem.h" // @manual
+
 #ifdef VELOX_ENABLE_ABFS
+#include "velox/common/base/Exceptions.h"
 #include "velox/common/config/Config.h"
 #include "velox/connectors/hive/storage_adapters/abfs/AbfsFileSystem.h" // @manual
 #include "velox/connectors/hive/storage_adapters/abfs/AbfsUtil.h" // @manual
+#include "velox/connectors/hive/storage_adapters/abfs/AzureClientProviderFactories.h" // @manual
+#include "velox/connectors/hive/storage_adapters/abfs/AzureClientProviders.h" // @manual
 #include "velox/dwio/common/FileSink.h"
 #endif
 
 namespace facebook::velox::filesystems {
 
 #ifdef VELOX_ENABLE_ABFS
+
+namespace {
+
+template <typename T>
+void registerProviderFactory(const std::string& accountName) {
+  AzureClientProviderFactories::registerFactory(
```

Comment:
This is a single-line code function. Let's inline the body below to make it easy to read.